portal/src/main/webapp/WEB-INF/jsp/dashboard/dashboard.jsp

```diff
@@ -285,22 +285,35 @@
             $("#study-tabs-loading-wait").css('display', 'none');
             if(_.isUndefined(window.iviz.datamanager)) {
                 window.iviz.datamanager = new DataManagerForIviz.init(window.cbioURL, studyCasesMap);
-                $.when(
-                    window.iviz.datamanager.initialSetup(),
-                    window.iviz.datamanager.getConfigs()
-                ).then(function(_data, configs) {
-                    var opts = {};
+                $.when(window.iviz.datamanager.getConfigs())
+                    .done(function(configs) {
+                        var opts = {};
 
-                    if (_.isObject(configs)) {
-                        opts = configs;
-                    }
-                    if (emailContact_) {
-                        opts.emailContact = emailContact_;
-                    }
-                    initdcplots(_data, opts);
-                });
-            }else {
-                $.when(window.iviz.datamanager.getConfigs()).then(function(configs){
+                        if (_.isObject(configs)) {
+                            opts = configs;
+                        }
+                        if (emailContact_) {
+                            opts.emailContact = emailContact_;
+                        }
+                        $.when(
+                            window.iviz.datamanager.initialSetup()
```

Comment:
this call used to be in parallel with `window.iviz.datamanager.getConfigs()` . Is it not possible to keep running both in parallel and go to "fail" is either one of them fails?

Reply:
The fail responses are different now. And since we are reading from a JSON file for the configs, so the performance wouldn't be affected.